### PR TITLE
Task/setup mongo backend

### PR DESCRIPTION
**Pull request by smyaseen**
## Goal:

Resolve issues in develop branch, backend API setup in front end, use observable for login and signup, save and get data from database, logout observable added, fixed warnings and errors.

## Work: 

All above done

### CODE REVIEW

File: a [hunk 1]
```diff
@@ -1,6 +0,0 @@
-* [32mdevelop[m
```
Comment (SiddiquiMonis401) [inline a:1]: What this a file does ?? This seems to be a useless file

File: src/Features/Auth/Login/index.js [hunk 2]
```diff
@@ -10,7 +10,6 @@ function LoginForm() {
   const dispatch = useDispatch();
 
   const loginClickHandler = () => {
-    console.log('asdsad');
     dispatch(login({ email: loginForm.email, password: loginForm.password }));
```
Comment (SiddiquiMonis401) [inline src/Features/Auth/Login/index.js:13]: As login feature has been integrated in development branch so try to use the input entered by the user

File: src/Features/Auth/actions.js [hunk 3]
```diff
@@ -1,8 +1,10 @@
-import { LOGIN, LOGIN_SUCCESS, LOGOUT, LOGIN_ERROR } from '../../scripts/constants';
+import { LOGIN, LOGIN_SUCCESS, LOGOUT, LOGIN_ERROR, SIGNUP } from '../../scripts/constants';
```
Comment (SiddiquiMonis401) [inline src/Features/Auth/actions.js:1]: It did not seem fit to place constants in scripts directory .. please put your ActionTypes in redux folder

File: src/Features/Auth/ducks.js [hunk 4]
```diff
@@ -1,41 +1,21 @@
...
-  LOGOUT,
-  LOGIN_ERROR,
-  SIGNUP,
-  SIGNUP_ERROR,
-  SIGNUP_SUCCESS,
-} from '../../scripts/constants';
+import { LOGIN, LOGIN_SUCCESS, LOGOUT, LOGIN_ERROR, SIGNUP } from '../../scripts/constants';
 
-export const loginReducer = (state = { isLoggedIn: false }, action) => {
+export const loginReducer = (state = { isLoggedIn: false, token: '', name: '' }, action) => {
```
Comment (SiddiquiMonis401) [inline src/Features/Auth/ducks.js:3]: As we are handling the whole auth logic in this reducer that name must be changed to auth reducer

File: src/routes/index.js [hunk 5]
```diff
@@ -18,12 +18,34 @@ export default function BaseRouter() {
...
-    <Switch>
-      {RouteConfig.auth.map((route, index) => {
-        return <Route key={index} component={() => <AuthRoute route={route} />} exact path={route.path} />;
-      })}
-      {isLoggedIn ? <Route component={Home} /> : <Route component={LoginContainer} />}
-    </Switch>
+    <>
+      <Switch>
+        {RouteConfig.auth.map((route, index) => {
+          return checkRoute(route, index);
```
Comment (SiddiquiMonis401) [inline src/routes/index.js:37]: Can we use some more meaningful name here instead if check route ??

File: src/routes/index.js [hunk 6]
```diff
@@ -18,12 +18,34 @@ export default function BaseRouter() {
...
+        })}
+        {isLoggedIn ? (
```
Comment (SiddiquiMonis401) [inline src/routes/index.js:39]: Why we are doing this ??

Comment (SiddiquiMonis401) [reply src/routes/index.js:39]: For dashboard routes and other private routes we must use the RouteConfigs array as we are using for auth

**Review by SiddiquiMonis401 — CHANGES_REQUESTED** [review]

**Review by ShahirZain — APPROVED** [review]

**Review by SiddiquiMonis401 — APPROVED** [review]

**Pull request merged**